test/src/data/article/ny_times_paginated_rest_repository_test.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:allthenews/src/data/article/ny_times_paginated_rest_repository.dart';
+import 'package:allthenews/src/data/communication/api/http_client.dart';
+import 'package:allthenews/src/domain/common/page.dart';
+import 'package:allthenews/src/domain/nytimes/ny_times_cached_repository.dart';
+import 'package:allthenews/src/domain/nytimes/ny_times_repository.dart';
+import 'package:allthenews/src/domain/settings/settings_repository.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+class MockHttpClient extends Mock implements HttpClient {}
+
+class MockSettingsRepository extends Mock implements SettingsRepository {}
+
+class MockNyTimesRepository extends Mock implements NYTimesRepository {}
+
+class MockNyTimesCachedRepository extends Mock implements NyTimesCachedRepository {}
+
+void main() {
+  NyTimesPaginatedRestRepository nyTimesPaginatedRestRepository;
+  MockNyTimesRepository mockNyTimesRepository;
+  MockNyTimesCachedRepository mockNyTimesCachedRepository;
+  MockSettingsRepository mockSettingsRepository;
+
+  setUp(() {
```

Comment:
to mogłoby się raz wykonywać przed wszystkimi testami, teraz się wykona przed każdą grupą, w pozostałych miejscach też można sprawdzić analogicznie

Reply:
ogólnie spoko, ale nie we wszystkich przypadkach, bo jednak czasami wraz z nową grupą jednak chce mieć czysty stan wejściowy :)